llvm/test/Instrumentation/RealtimeSanitizer/rtsan.ll

```diff
@@ -0,0 +1,35 @@
+; RUN: opt < %s -passes=rtsan -S | FileCheck %s
+
+; Function Attrs: mustprogress noinline nonblocking optnone ssp uwtable(sync)
+define void @violation() #0 {
+  %1 = alloca ptr, align 8
+  %2 = call ptr @malloc(i64 noundef 2) #3
+  store ptr %2, ptr %1, align 8
+  ret void
+}
+
+; Function Attrs: allocsize(0)
+declare ptr @malloc(i64 noundef) #1
+
+; Function Attrs: mustprogress noinline norecurse optnone ssp uwtable(sync)
+define noundef i32 @main() #2 {
+  %1 = alloca i32, align 4
+  store i32 0, ptr %1, align 4
+  call void @violation() #4
+  ret i32 0
+}
+
+attributes #0 = { mustprogress noinline nonblocking optnone ssp uwtable(sync) }
+attributes #1 = { allocsize(0) }
+attributes #2 = { mustprogress noinline norecurse optnone ssp uwtable(sync) }
+attributes #3 = { allocsize(0) }
+attributes #4 = { nonblocking }
+
+; RealtimeSanitizer pass should insert __rtsan_realtime_enter right after function definition
+; CHECK: define{{.*}}violation
+; CHECK-NEXT: call{{.*}}__rtsan_realtime_enter
+
+; RealtimeSanitizer pass should insert __rtsan_realtime_exit right before function return
+; CHECK: call{{.*}}@__rtsan_realtime_exit
+; CHECK-NEXT: ret{{.*}}void
```

Comment:
One question we had: We want to be able to insert this before EVERY function return. However I was not able to generate an example that ever had multiple return statements in llvm-ir.

Is it possible to have multiple return statements in IR?
If so, what is a C/C++ example that induces this?

Adding a test to ensure it is added before every return statement would be a great improvement.

Reply:
> Is it possible to have multiple return statements in IR?
Yes

> If so, what is a C/C++ example that induces this?
Not sure.

> Adding a test to ensure it is added before every return statement would be a great improvement.
You can just write that IR whithout c++

Reply:
I will investigate and add this to the other review if I crack it!! Thanks for the hint.

This review is now just the attribute, I'll move these comments over there.